src/openrct2/object/RideObject.cpp

```diff
@@ -43,6 +43,55 @@ static const uint8_t SpriteGroupMultiplier[EnumValue(SpriteGroupType::Count)] =
     1, 2, 2, 2, 2, 2, 2, 10, 1, 2, 2, 2, 2, 2, 2, 2, 6, 4, 4, 4, 4, 4, 4, 4, 12, 4, 4, 4, 4, 4, 20, 3, 1,
 };
 
+constexpr const uint8_t DefaultSteamSpawnPosition[] = { 11, 22 };
+
+static const EnumMap<CarEntryAnimation> AnimationNameLookup{
+    { "none", CarEntryAnimation::None },
+    { "simpleVehicle", CarEntryAnimation::SimpleVehicle },
+    { "steamLocomotive", CarEntryAnimation::SteamLocomotive },
+    { "swanBoat", CarEntryAnimation::SwanBoat },
+    { "monorailCycle", CarEntryAnimation::MonorailCycle },
+    { "MultiDimension", CarEntryAnimation::MultiDimension },
+    { "observationTower", CarEntryAnimation::ObservationTower },
+    { "animalFlying", CarEntryAnimation::AnimalFlying },
+};
+
+constexpr const auto NumLegacyAnimationTypes = 11;
+
+struct LegacyAnimationParameters
+{
+    uint16_t Speed;
+    uint8_t NumFrames;
+    CarEntryAnimation Alias;
+};
+
+constexpr const LegacyAnimationParameters VehicleEntryDefaultAnimation[NumLegacyAnimationTypes] = {
+    { 0, 1, CarEntryAnimation::None },                  // None
+    { 1 << 12, 4, CarEntryAnimation::SteamLocomotive }, // Miniature Railway Locomotive
+    { 1 << 10, 2, CarEntryAnimation::SwanBoat },        // Swan Boat
+    { 1 << 11, 6, CarEntryAnimation::SimpleVehicle },   // Canoe
+    { 1 << 11, 7, CarEntryAnimation::SimpleVehicle },   // Rowboat
+    { 1 << 10, 2, CarEntryAnimation::SimpleVehicle },   // Water Tricycle
+    { 0x3333, 8, CarEntryAnimation::ObservationTower }, // Observation Tower
+    { 1 << 10, 4, CarEntryAnimation::SimpleVehicle },   // Mini Helicopter
+    { 1 << 11, 4, CarEntryAnimation::MonorailCycle },   // Monorail Cycle
+    { 0x3333, 8, CarEntryAnimation::MultiDimension },   // Multi Dimension Coaster
+    { 24, 4, CarEntryAnimation::AnimalFlying },         // Animal Flying
+};
```

Comment:
Also use static_assert to verify the size here, and leave `NumLegacyAnimationTypes` out of the brackets.

```suggestion
constexpr const LegacyAnimationParameters VehicleEntryDefaultAnimation[] = {
    { 0, 1, CarEntryAnimation::None },                  // None
    { 1 << 12, 4, CarEntryAnimation::SteamLocomotive }, // Miniature Railway Locomotive
    { 1 << 10, 2, CarEntryAnimation::SwanBoat },        // Swan Boat
    { 1 << 11, 6, CarEntryAnimation::SimpleVehicle },   // Canoe
    { 1 << 11, 7, CarEntryAnimation::SimpleVehicle },   // Rowboat
    { 1 << 10, 2, CarEntryAnimation::SimpleVehicle },   // Water Tricycle
    { 0x3333, 8, CarEntryAnimation::ObservationTower }, // Observation Tower
    { 1 << 10, 4, CarEntryAnimation::SimpleVehicle },   // Mini Helicopter
    { 1 << 11, 4, CarEntryAnimation::MonorailCycle },   // Monorail Cycle
    { 0x3333, 8, CarEntryAnimation::MultiDimension },   // Multi Dimension Coaster
    { 24, 4, CarEntryAnimation::AnimalFlying },         // Animal Flying
};
static_assert(std::size(VehicleEntryDefaultAnimation) == NumLegacyAnimationTypes);
```